### ISSUE TITLE
`--chunk-size` with selector returns missing result, so `describe node` / `drain` could be works wrong

### ISSUE THREAD

**Issue by yokomotod** [kubernetes/kubernetes#110625]
### What happened?

`kubectl get` has `--chunk-size=500` by default, and since 1.22 `describe` `drain` has the same.   
 https://github.com/kubernetes/kubectl/issues/1028

They seem to return wrong results in certain situations combined with `--selector` / `--field-selector`.

And `kubectl describe node` / `kubectl drain` also use `--field-selector=spec.nodeName=NODE`, so they also don't work correctly.  
`describe node` outputs partial list of pods on the node, `drain` doesn't evict some of the pods.

To make it work correctly, I had to specify `--chunk-size=0`

### What did you expect to happen?

`get` `describe` `drain` works correctly with any `chunk-size`, since it's just a performance optimizing parameter.

### How can we reproduce it (as minimally and precisely as possible)?

I can't provide an actual concrete example (no `--chunk-size` = default `chunk-size=500`), but instead I show how to reproduce it with a smaller `chunk-size`

I believe that the same problem occurs with `chunk-size=500`

----

create two deployments.

```
$ kubectl create deployment hello-server --image=us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0 --replicas=5
$ kubectl create deployment another-server --image=us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0 --replicas=5
```

 there are 5 pods in each.

```
$ kubectl get po
NAME                              READY   STATUS    RESTARTS   AGE
another-server-65f84969d6-h99mt   1/1     Running   0          4s
another-server-65f84969d6-hfc96   1/1     Running   0          4s
another-server-65f84969d6-lm26l   1/1     Running   0          4s
another-server-65f84969d6-vk44c   1/1     Running   0          4s
another-server-65f84969d6-x6vl5   1/1     Running   0          4s
hello-server-7bb955f957-db4sc     1/1     Running   0          58s
hello-server-7bb955f957-dsdpj     1/1     Running   0          58s
hello-server-7bb955f957-hdw8m     1/1     Running   0          58s
hello-server-7bb955f957-x9m4t     1/1     Running   0          58s
hello-server-7bb955f957-z44n2     1/1     Running   0          58s
```

with --selector=app=another-server and `--chunk-size=2`, it shows only 4 pods.

```
$ kubectl get po --selector=app=another-server --chunk-size=2
NAME                              READY   STATUS    RESTARTS   AGE
another-server-65f84969d6-h99mt   1/1     Running   0          10s
another-server-65f84969d6-hfc96   1/1     Running   0          10s
another-server-65f84969d6-lm26l   1/1     Running   0          10s
another-server-65f84969d6-vk44c   1/1     Running   0          10s
```

### Anything else we need to know?

I could confirm that the problem happens

* in both  case of `--selector` and  `--field-selector`
* regardless namespace, even with `-n kube-system`, or `--all-namespaces`

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:49:13Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6-gke.1700", GitCommit:"694183800acb7cce57699639c06a9dd196c9f227", GitTreeState:"clean", BuildDate:"2022-05-13T09:31:28Z", GoVersion:"go1.17.9b7", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
GCP / GKE Standard
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by Abirdcfly** [comment]
I think this is indeed a bug, after some simple testing, it seems to be an bug unique to 1.24, let me fix it. /assign

**Comment by Abirdcfly** [comment]
I can reproduce this error in a 1.24 cluster, but not in a 1.23.6 cluster. All clusters are created via kind, and I'm guessing gke may have cherry-picked some commits or had some of their own changes to their branch. 

So I think this fix should at least cherry-pick to 1.23 to help other distributions fix this issue.

`kind create cluster  --image kindest/node:v1.23.6`
